packages/syft/src/syft/node/node.py

```diff
@@ -904,59 +839,65 @@ def worker_stash(self) -> WorkerStash:
-            kwargs = {}
-            if service_klass == ActionService:
-                kwargs["store"] = self.action_store
-            store_services = [
-                UserService,
-                WorkerService,
-                SettingsService,
-                DatasetService,
-                UserCodeService,
-                LogService,
-                RequestService,
-                QueueService,
-                JobService,
-                DataSubjectService,
-                NetworkService,
-                PolicyService,
-                NotifierService,
-                NotificationService,
-                DataSubjectMemberService,
-                ProjectService,
-                EnclaveService,
-                CodeHistoryService,
-                MetadataService,
-                BlobStorageService,
-                MigrateStateService,
-                SyftWorkerImageService,
-                SyftWorkerPoolService,
-                SyftImageRegistryService,
-                SyncService,
-                OutputService,
-                UserCodeStatusService,
-            ]
-
-            if OBLV:
-                # relative
-                from ..external.oblv.oblv_service import OblvService
+        service_path_map: dict[str, AbstractService] = {}
+        initialized_services: list[AbstractService] = []
+
+        # A dict of service and init kwargs.
+        # - "svc" expects a callable (class or function)
+        #     - The callable must return AbstractService or None
+        # - "store" expects a store type
+        #     - By default all services get the document store
+        #     - Pass a custom "store" to override this
+        default_services: list[dict] = [
+            {"svc": ActionService, "store": self.action_store},
+            {"svc": UserService},
+            {"svc": WorkerService},
+            {"svc": SettingsService},
+            {"svc": DatasetService},
+            {"svc": UserCodeService},
+            {"svc": LogService},
+            {"svc": RequestService},
+            {"svc": QueueService},
+            {"svc": JobService},
+            {"svc": DataSubjectService},
+            {"svc": NetworkService},
+            {"svc": PolicyService},
+            {"svc": NotifierService},
+            {"svc": NotificationService},
+            {"svc": DataSubjectMemberService},
+            {"svc": ProjectService},
+            {"svc": EnclaveService},
+            {"svc": CodeHistoryService},
+            {"svc": MetadataService},
+            {"svc": BlobStorageService},
+            {"svc": MigrateStateService},
+            {"svc": SyftWorkerImageService},
+            {"svc": SyftWorkerPoolService},
+            {"svc": SyftImageRegistryService},
+            {"svc": SyncService},
+            {"svc": OutputService},
+            {"svc": UserCodeStatusService},
+            {"svc": VeilidServiceProvider},  # this is lazy
+            {"svc": OblvServiceProvider},  # this is lazy
```

Comment:
With the provider pattern, we can defer initialization of services to the module. This way we don't need to manually place if-else blocks